### PR TITLE
Ompi master

### PR DESCRIPTION
**Pull request by avildema**

### CODE REVIEW

File: slurm_deploy/deploy_ctl.sh [hunk 1]
```diff
@@ -80,104 +71,119 @@ function item_download() {
...
                 exit 1
-        	fi
-        	echo "tar_opts = $tar_opts"
-            if [ ${packurl:0:1} = "/" ]; then
+            fi
+            echo "tar_opts = $tar_opts"
+            if [ "${packurl:0:1}" = "/" ]; then
                 echo "unpacking \"$REPO_NAME\" from local path..."
-                cat $packurl | tar $tar_opts -C $SRC_DIR/$REPO_NAME --strip-components 1
+                # TODO
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:125]: What exactly should be TODO? Please add comments.

File: slurm_deploy/deploy_ctl.sh [hunk 2]
```diff
@@ -80,104 +71,119 @@ function item_download() {
...
+                # shellcheck disable=SC2002
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:126]: What means this comment? Let's remove it.

File: slurm_deploy/deploy_ctl.sh [hunk 3]
```diff
@@ -80,104 +71,119 @@ function item_download() {
...
         if [ -n "$commit" ]; then
-            cd $REPO_SRC
-            git checkout -b test $commit
+            cd "$REPO_SRC"
+            git checkout -b test "$commit"
             cd -
         fi
     fi
 
+    # shellcheck disable=SC2181
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:145]: What means this comment? Let's remove it.

File: slurm_deploy/deploy_ctl.sh [hunk 4]
```diff
@@ -80,104 +71,119 @@ function item_download() {
...
 
     if [ -n "$config" ]; then
         echo "\"$REPO_NAME\": the following config will be configure : \"$config\""
     fi
 
-    
-# create the configure script for we can configure it later
-    cat > $build/config.sh << EOF
+    if [ "${REPO_NAME}" = "ompi" ]; then
+        CONFIGURE_EXTRA_ARGS="${OMPI_CONFIGURE_EXTRA_ARGS}"
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:164]: `CONFIGURE_EXTRA_ARGS` and `OMPI_CONFIGURE_EXTRA_ARGS` this vars are not used, let's delete it.

File: slurm_deploy/deploy_ctl.sh [hunk 5]
```diff
@@ -213,58 +219,66 @@ function deploy_source_prepare() {
     item_download "ucx" "$UCX_PACK" "$UCX_URL" "$UCX_INST" "$UCX_BRANCH" "$UCX_COMMIT" "$UCX_CONF"
     deploy_item_save_env "$REPO_NAME" "$REPO_INST" "$REPO_SRC" "UCX"
 
+    # shellcheck disable=SC2153
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:222]: What means this comment? Let's remove it.

File: slurm_deploy/deploy_ctl.sh [hunk 6]
```diff
@@ -213,58 +219,66 @@ function deploy_source_prepare() {
...
+    item_download "ompi" "$OMPI_PACK" "$OMPI_URL" "$OMPI_INST" "$OMPI_BRANCH" "$OMPI_COMMIT" \
+ "--with-pmix=$PMIX_INST --with-slurm=$SLURM_INST --with-libevent=$LIBEV_INST --with-ucx=$UCX_INST --with-hwloc=$HWLOC_INST $OMPI_CONF"
     deploy_item_save_env "$REPO_NAME" "$REPO_INST" "$REPO_SRC" "OMPI"
 }
 
 function get_item() {
     item_inst=$1
-    item=`cat $DEPLOY_DIR/.deploy_repo.lst | grep $item_inst | awk '{print $1}'`
-    echo $item
+    # TODO
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:234]: The same above. Add comment.

File: slurm_deploy/deploy_ctl.sh [hunk 7]
```diff
@@ -213,58 +219,66 @@ function deploy_source_prepare() {
...
+    # shellcheck disable=SC2002
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:235]: remove

File: slurm_deploy/deploy_ctl.sh [hunk 8]
```diff
@@ -213,58 +219,66 @@ function deploy_source_prepare() {
...
+    item=$(cat "$DEPLOY_DIR/.deploy_repo.lst" | grep "$item_inst" | awk '{print $1}')
+    echo "$item"
 }
 
 function get_repo_item_lst() {
-    repo_items=`cat $DEPLOY_DIR/.deploy_repo.lst | awk '{print $2}'`
-    echo $repo_items
+    # shellcheck disable=SC2002
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:241]: remove

File: slurm_deploy/deploy_ctl.sh [hunk 9]
```diff
@@ -273,302 +287,317 @@ function deploy_build_item() {
                 export PATH=$tools_path:$PATH && ./autogen.pl
             fi
         fi
+
+        # shellcheck disable=SC2181
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:291]: remove

File: slurm_deploy/deploy_ctl.sh [hunk 10]
```diff
@@ -273,302 +287,317 @@ function deploy_build_item() {
...
             exit 1
         fi
     fi
+
     cd .build || (echo_error $LINENO "directory change error" && exit 1)
+
     if [ ! -f "config.log" ]; then
-        pdsh -S -w $build_node "cd $PWD && ./config.sh"
+        pdsh -S -w "$build_node" "cd $PWD && LD_LIBRARY_PATH=${HWLOC_INST}/lib:${LIBEV_INST}/lib:${PMIX_INST}/lib:${LD_LIBRARY_PATH} ./config.sh"
+        # shellcheck disable=SC2181
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:303]: remove

File: slurm_deploy/deploy_ctl.sh [hunk 11]
```diff
@@ -273,302 +287,317 @@ function deploy_build_item() {
...
             exit 1
         fi
-        echo 1 > .deploy_build_flag
+        echo 1 >.deploy_build_flag
     fi
-    pdsh -S -w $build_node "cd $PWD && make -j $build_cpus install"
+
+    pdsh -S -w "$build_node" "cd $PWD && make -j $build_cpus install"
     ret=$?
+    # shellcheck disable=SC2181
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:323]: remove

File: slurm_deploy/deploy_ctl.sh [hunk 12]
```diff
@@ -273,302 +287,317 @@ function deploy_build_item() {
...
-         pdsh -S -w $build_node "cd $PWD/contribs/pmi2 && make -j $build_cpus install"
+    if [ "$item" = "slurm" ]; then
+        pdsh -S -w "$build_node" "cd $PWD/contribs/pmi && make -j $build_cpus install"
+        pdsh -S -w "$build_node" "cd $PWD/contribs/pmi2 && make -j $build_cpus install"
+    fi
+
+    if [ "$item" = "ompi" ]; then
+        # TODO: tmp workaround
+        echo "INFO: $SRC_DIR/$item/prrte/src/include/prrte_config.h will be removed"
+        rm -f "$SRC_DIR/$item/prrte/src/include/prrte_config.h"
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:338]: Please remove this workaround. The issue with `prrte_config.h` was fixed with https://github.com/open-mpi/ompi/pull/7667.

File: slurm_deploy/deploy_ctl.sh [hunk 13]
```diff
@@ -273,302 +287,317 @@ function deploy_build_item() {
...
-    make -j $CPU_NUM install
-    for file in `ls $SLURM_INST/lib/slurm/mpi_pmix*`; do
-        copy_remote_nodes $nodes $file $SLURM_INST/lib/slurm/
+    sdir=$(pwd)
+    nodes=$(distribute_get_nodes)
+    item=$(get_item "$SLURM_INST")
+    cd "$SRC_DIR/$item/.build/src/plugins/mpi/pmix"
+    make -j "$CPU_NUM" clean
+    make -j "$CPU_NUM" install
+    # shellcheck disable=SC2045
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:400]: remove

File: slurm_deploy/deploy_ctl.sh [hunk 14]
```diff
@@ -273,302 +287,317 @@ function deploy_build_item() {
...
-    first_node=`hostname`
+    distribute_nodes=$(distribute_get_nodes) # nodes on which the software will be distributed
+    first_node=$(hostname)
     if [ -n "$distribute_nodes" ]; then
-        first_node=`scontrol show hostname $distribute_nodes | head -n 1` # get first node for run build on it
+        first_node=$(scontrol show hostname "$distribute_nodes" | head -n 1) # get first node for run build on it
     fi
-    slurm_ctl_node=`ssh $first_node cat $SLURM_INST/etc/local.conf | grep ControlMachine | cut -f2 -d"="`
-    exec_remote_as_user_nodes $slurm_ctl_node $SLURM_INST/sbin/slurmctld
+    # shellcheck disable=SC2029
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:517]: remove

File: slurm_deploy/deploy_ctl.sh [hunk 15]
```diff
@@ -273,302 +287,317 @@ function deploy_build_item() {
...
-            sed -e "s/@node_sock_num@/$SOCKETS/g" | \
-            sed -e "s/@node_core_per_socket@/$CORE_PER_SOCK/g" | \
-            sed -e "s/@node_thread_per_core@/$THREAD_PER_CORE/g" | \
-            sed -e "s/@node_list@/$CFG_NODE_LIST/g" | \
-            sed -e "s/@partition@/$SLURM_JOB_PARTITION/g" | \
-            sed -e "s/@node_ctl@/$CONTROL_MACHINE/g" > $tdir/local.conf
-
-        cp $tdir/local.conf $SLURM_INST/etc/
+        #generate a config file
+        # shellcheck disable=SC2002
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:567]: remove

File: slurm_deploy/deploy_ctl.sh [hunk 16]
```diff
@@ -273,302 +287,317 @@ function deploy_build_item() {
...
+        cp $tdir/local.conf "$SLURM_INST/etc/"
         rm -fR $tdir
     fi
 
-    SLURM_INST_ESC=`escape_path $SLURM_INST`
-    cat $FILES/slurm.conf.in | \
-        sed -e "s/@SLURM_INST@/$SLURM_INST_ESC/g" | \
-        sed -e "s/@SLURM_USER@/$SLURM_USER/g" > $SLURM_INST/etc/slurm.conf
+    SLURM_INST_ESC=$(escape_path "$SLURM_INST")
+    # shellcheck disable=SC2002
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:583]: remove

File: slurm_deploy/deploy_ctl.sh [hunk 17]
```diff
@@ -80,104 +71,119 @@ function item_download() {
...
+                # shellcheck disable=SC2002
+                cat "$packurl" | tar $tar_opts -C "$SRC_DIR/$REPO_NAME" --strip-components 1
             else
-                curl -L $packurl | tar $tar_opts -C $SRC_DIR/$REPO_NAME --strip-components 1
+                curl -L "$packurl" | tar $tar_opts -C "$SRC_DIR/$REPO_NAME" --strip-components 1
             fi
             if [ "0" -ne "${PIPESTATUS[0]}" ]; then
                 echo_error $LINENO "\"$REPO_NAME\" Repository can not be obtained. Cannot continue. "
-                rm -rf $SRC_DIR/$REPO_NAME
+                rm -rf "${SRC_DIR:?}/$REPO_NAME"
```
Comment (karasevb) [inline slurm_deploy/deploy_ctl.sh:133]: `:?` - what benefits of this change?

File: slurm_deploy/deploy_ctl.sh (hunk 4 above)
Comment (avildema) [reply slurm_deploy/deploy_ctl.sh:164]: Used here

`# create the configure script for we can configure it later
    cat >"$build/config.sh" <<EOF
#!/bin/bash

echo "INFO: \$PATH"
echo "INFO: \$LD_LIBRARY_PATH"
echo "INFO: $REPO_SRC/configure ${CONFIGURE_EXTRA_ARGS} $config"

$REPO_SRC/configure ${CONFIGURE_EXTRA_ARGS} $config
EOF`

Comment (karasevb) [reply slurm_deploy/deploy_ctl.sh:160]: Actually, the `OMPI_CONFIGURE_EXTRA_ARGS` is redundant variable, all extra args may be obtained thought `$config`

**Comment by avildema** [comment]
move to  https://github.com/Mellanox-lab/jobstart/

**Pull request closed without merging**